>From a05980aeeefed3a5f6e68bd8c9b37124ecdb8f03 Mon Sep 17 00:00:00 2001
From: Joel Rosdahl <joel@rosdahl.net>
Date: Sun, 9 Aug 2015 22:07:11 +0200
Subject: [PATCH] Exit properly from exit handler

This fixes a problem where two SIGINT (Ctrl+C) signals were needed to
terminate an ongoing compilation. (The first SIGINT terminated the
compiler which exited with non-zero status, which ccache then went on
interpreting as an ordinary failed compilation, leading to rerunning the
compiler via failed().)
---
 ccache.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ccache.c b/ccache.c
index f0dca22..de63cfe 100644
--- a/ccache.c
+++ b/ccache.c
@@ -264,6 +264,7 @@ signal_handler(int signo)
 {
 	(void)signo;
 	clean_up_pending_tmp_files();
+	_exit(1);
 }
 
 static void
-- 
2.1.4

